### DISCUSSION TITLE
how to name widgets in modules in contrib?

### DISCUSSION THREAD

**Discussion by bkerin** [akinomyoga/ble.sh#330] [General]
At present I'm following the advice I remember seeing somewhere in the user docs (I can't find it at the moment) and name my widgets like for example: 

     ble/widget/my/vi-command/forward-my-vword

IIRC they were names this way so some command that knows about widgets could find them, and so they could be named compactly when given to e.g. ble-bind (withteh ble/widget prefix omitted).

I guess my existing name should be changed to:

     ble/widget/contrib/config:vim-wordmotion-ish/vi-command/forward-vword

but the style guide doesn't explicitly say how these should be named?

**Comment by akinomyoga** [comment]
> At present I'm following the advice I remember seeing somewhere in the user docs (I can't find it at the moment)

It's in the `style` branch of PR #317.

> name my widgets like for example:
> 
> ```bash
>  ble/widget/my/vi-command/forward-my-vword
> ```

Yes.

> I guess my existing name should be changed to:
> 
> ```
>  ble/widget/contrib/config:vim-wordmotion-ish/vi-command/forward-vword
> ```
> 
> but the style guide doesn't explicitly say how these should be named?

Ah, yes. It's not specified in the style guide. Since `contrib/config:vim-wordmotion-ish/vi-command/forward-vword` is too long when specified to `ble-bind`, I think the name should be shorter, but it's difficult to make a strict rule. Currently, I'd like to determine it in each PR. Since we anyway discuss it in a PR when necessary, you do not have to worry about it too much for the first version of PR. You can first name them as you like, and then we can discuss and adjust them.

Actually, the primary reason that we use namespaces in ble.sh is to avoid possible conflicts. For the `contrib` repository, as it is maintained as a part of `ble.sh`, the problem of name conflicts is less significant. In particular, the widget names can be more compact because they are a part of the exposed interface and we want shorter names there. It's different from the case of functions internally used in the defined module.

My current preference would be something like `ble/widget/vim-wordmotion-nish/forward-vword`.
